mne/annotations.py

```diff
@@ -101,11 +101,10 @@ def _combine_annotations(annotations, last_samps, first_samps, sfreq):
         old_description = annotations[0].description
         old_orig_time = annotations[0].orig_time
 
-    if annotations[1].orig_time is None:
-        onset = (annotations[1].onset +
-                 (sum(last_samps[:-1]) - sum(first_samps[:-1])) / sfreq)
-    else:
-        onset = annotations[1].onset
+    extra_samps = len(first_samps) - 1  # Account for sample 0
```

Comment:
Previously the annotations were shifting by 1 sample when concatenated. This fixes it.